user/config.go

```diff
@@ -0,0 +1,9 @@
+package user
+
+// Config contains the user's configuration.
+type Config struct {
```

Comment:
I made a `user.Config` that I think (hope) will attract some of the floating behavior related to home directory, config directory, etc.

I realized that we have two different types of configuration: configuration related to the user (their workspace, their config file, their home directory, their api token), and api configuration (endpoints which are hardcoded at the moment, and hosts which have defaults).

In nextercism we're going to add a third type: cli-specific configuration, e.g. the various error messages and templates and strings for communicating with the user.